src/compiler/crystal/syntax/ast.cr

```diff
@@ -5,9 +5,31 @@ module Crystal
     # if the location is not known.
     property location : Location?
 
+    # The line and column where the node ends.
+    # The file where the node ends is always that of
+    # `location`.
+    property end_line_and_column = LineAndColumn.empty
```

Comment:
Why not make this nilable? If we do that, this will occupy 16 bytes instead of 8. So doing it this way is a compromise.